src/main/kotlin/com/assisteddarwinism/ultimatesmashtracker/player/PlayerController.kt

```diff
@@ -20,32 +25,61 @@ class PlayerController {
     @Autowired
     lateinit var playerRepository: PlayerRepository
 
+    @Autowired
+    lateinit var tokenRepository: TokenRepository
+
+    @Autowired
+    lateinit var matchRepository: MatchRepository
+
+    @Autowired
+    lateinit var playerCharacterCombinationRepository: PlayerCharacterCombinationRepository
+
     @GetMapping
-    fun getPlayers(@RequestHeader("X-AuthToken") token: String) : List<Player> {
+    fun getPlayers(@RequestHeader("X-AuthToken") token: String): List<Player> {
         tokenValidator.checkTokenValid(token)
 
         return playerRepository.findAll().map { it -> Player(it.id!!, it.name) }
     }
 
     @GetMapping("/{playerId}")
-    @ResponseStatus(HttpStatus.NOT_IMPLEMENTED)
-    fun getPlayer(@RequestHeader("X-AuthToken") token: String, @PathVariable("playerId") playerId: Long){
+    fun getPlayerDetails(@RequestHeader("X-AuthToken") token: String, @PathVariable("playerId") playerId: Long): Player {
         tokenValidator.checkTokenValid(token)
+        if (!playerRepository.existsById(playerId)) throw ResourceNotFoundException()
+        val player = playerRepository.findById(playerId).get()
+        val matches = matchRepository.findAll()
+                .map { Match(it, playerCharacterCombinationRepository.findAllByMatchId(it.id!!)
+                        .map { PlayerCharacterRelation(it) }) }
+
+        return if (matches.isNotEmpty()) {
+            val wins = matches.count { it.winner == player.id }
+            val losses = wins - matches.size
+            val mostPlayedCharId = matches.flatMap { it.players }
+                    .filter { it.playerId == player.id }
+                    .groupBy { it.characterId }
+                    .maxBy { it.value.size }!!.key
+            val mostRecentMatchId = matches.maxBy { it.time }!!.id
+            Player(player.id!!, player.name, wins, losses, mostPlayedCharId, mostRecentMatchId)
+        } else {
+            Player(player.id!!, player.name)
+        }
     }
 
-    @GetMapping("/{playerId}/matches")
-    @ResponseStatus(HttpStatus.NOT_IMPLEMENTED)
-    fun getPlayerMatches(@RequestHeader("X-AuthToken") token: String, @PathVariable("playerId") playerId: Long){
+    @GetMapping("/me")
+    fun getMe(@RequestHeader("X-AuthToken") token: String): Player {
         tokenValidator.checkTokenValid(token)
+        var currentUserId = tokenRepository.findTokenDTOByToken(token).id

```

Comment:
Maybe we should do this in tokenValidator.   Hand it a token, get an id as a result.  That way if we change how tokens work in the future the changes are constrained to a single class.

It would also let us re-use it anywhere else you want to support using "me" as a playerId